### PR TITLE
Autogenerate DB schema HTML docs with polymorphic associations

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves limitation of DB schema documentation generators like [SchemaSpy](http://schemaspy.org/) ([generated Caseflow schema doc](https://htmlpreview.github.io/?https://github.com/yoomlam/generated_docs/blob/master/schemaspy/output/index.html)) or https://dbdiagram.io (used for creating diagrams at https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary).
The limitation being that (1) Rails polymorphic associations are not reflected in the Postgres database configuration/setup/DDL and (2) there are a lot of tables to fit in a single visualization.

### Description
This PR extracts polymorphic associations into a file that can be fed to [Jailer](https://github.com/Wisser/Jailer) to create html pages that show each table's associations (and SQL `join on` arguments!):
For example, associations for `LegacyAppeal`:
image

The autogenerated html files are not included in this PR and will be created by a GitHub Action and posted to http://department-of-veterans-affairs.github.io/caseflow/schema/index.html. 

This will be helpful for anyone creating SQL queries, particularly the Reporting Team, esp. when used as accompaniment with [these ERD diagrams](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Database-Schema-Documentation#caseflow-erd).

This is how the html would look using `htmlpreview.github.io`: https://htmlpreview.github.io/?https://github.com/yoomlam/generated_docs/blob/master/jailer/index.html
Or we can enable [GitHub Pages](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site) and view it like [this](https://yoomlam.github.io/generated_docs/jailer/).

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] polymorphic associations are extracted into a file
- [ ] html pages showing Rails model associations

### Testing Plan
```
SCHEMA=caseflow bundle exec rake doc:belongs_to:jailer_polymorphic_associations
# creates docs/schema/caseflow-jailer_polymorphic_associations.csv
```
(This is also run as a result of running `make docs`.)

Refer to instructions in [this code comment](https://github.com/department-of-veterans-affairs/caseflow/pull/16657/files?file-filters%5B%5D=.rake&file-filters%5B%5D=.rb#diff-faf9971a389ff22a7638216daf4683d292ce419fe84517cdcbd8451577feb782R197).

### CODE REVIEW

File: docs/schema/caseflow.csv [hunk 1]
```diff
@@ -1242,6 +1242,13 @@ vso_configs,id,integer (8) PK,x,x,,,,
 vso_configs,ihp_dockets,string,,,,,,
 vso_configs,organization_id,integer FK,,,x,,x,
 vso_configs,updated_at,datetime ∗,x,,,,x,
+work_modes,,,,,,,,Captures user's current work mode for appeals being worked
+work_modes,appeal_id,integer ∗ FK,x,,x,,x,Appeal ID -- use as FK to AMA appeals and legacy appeals
+work_modes,appeal_type,string ∗,x,,,,x,Whether appeal_id is for AMA or legacy appeals
+work_modes,created_at,datetime ∗,x,,,,,Standard created_at/updated_at timestamps
+work_modes,id,integer (8) PK,x,x,,,,
+work_modes,overtime,boolean,,,,,,Whether the appeal is currently marked as being worked as overtime
+work_modes,updated_at,datetime ∗,x,,,,,Standard created_at/updated_at timestamps
```
Comment (yoomlam) [inline docs/schema/caseflow.csv:1251]: Artifact of fixing the parent class of the `WorkMode` model and running `make docs`.

File: app/models/work_mode.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # Model to represent the mode in which a user (e.g., an attorney) is working on a case.
 # Specifically, the `overtime` field captures whether the appeal is being worked as overtime for the attorney.
 
-class WorkMode < ApplicationRecord
+class WorkMode < CaseflowRecord
```
Comment (yoomlam) [inline app/models/work_mode.rb:6]: Fix: this should have been `CaseflowRecord`.

**Comment by yoomlam** [comment]
You can skip review of the `html` files since they are autogenerated:
image

**Comment by yoomlam** [comment]
I'm going to convert this into a GitHub Action during the IP sprint.

**Comment by yoomlam** [comment]
This PR can be reviewed and merged. I'll create the GH Action and update the code comments in another PR.

**Review by lowellrex — APPROVED** [review]
LGTM!

**Pull request merged**